src/components/capture/AutoDiscoverySettings.tsx

```diff
@@ -0,0 +1,255 @@
+                autoDiscoverySettings.addNewBindings;
+
+            const evolutionSetting: boolean =
+                Object.hasOwn(
+                    autoDiscoverySettings,
+                    'evolveIncompatibleCollections'
+                ) && autoDiscoverySettings.evolveIncompatibleCollections;
+
+            setAddNewBindings(bindingSetting, {
+                initOnly: true,
+            });
+
+            setEvolveIncompatibleCollections(evolutionSetting, {
+                initOnly: true,
+            });
+        } else {
+            setAddNewBindings(false, { initOnly: true });
+            setEvolveIncompatibleCollections(false, { initOnly: true });
+        }
+    }, [
+        setAddNewBindings,
+        setAutoDiscover,
+        setEvolveIncompatibleCollections,
+        draftSpecs,
+        settingsExist,
+    ]);
+
+    const serverUpdateRequired = useMemo(() => {
+        const settingsNew =
+            !settingsExist &&
+            (autoDiscover || addNewBindings || evolveIncompatibleCollections);
+
+        const settingsChanged =
+            settingsExist &&
+            (!autoDiscover ||
+                !isEqual(draftSpecs[0].spec.autoDiscover, {
+                    addNewBindings,
+                    evolveIncompatibleCollections,
+                }));
+
+        return settingsNew || settingsChanged;
+    }, [
+        addNewBindings,
+        autoDiscover,
+        draftSpecs,
+        evolveIncompatibleCollections,
+        settingsExist,
+    ]);
+
+    useEffect(() => {
+        if (settingsSaving) {
+            if (serverUpdateRequired) {
+                setFormState({ status: FormStatus.UPDATING });
+
+                updateAutoDiscoverySettings()
+                    .then(
+                        () => setFormState({ status: FormStatus.UPDATED }),
+                        (error) =>
+                            setFormState({ status: FormStatus.FAILED, error })
+                    )
+                    .finally(() => setSettingsSaving(false));
+            } else {
+                setSettingsSaving(false);
+            }
+        }
+    }, [
+        setFormState,
+        setSettingsSaving,
+        updateAutoDiscoverySettings,
+        serverUpdateRequired,
+        settingsSaving,
+    ]);
+
+    const containerEl = document.getElementById(containerId);
+
+    useEffect(() => {
+        let resizeObserver: ResizeObserver | null = null;
+
+        if (containerEl) {
+            resizeObserver = new ResizeObserver(() => {
+                setDisplayDirection(
+                    containerEl.clientWidth < 600 ? 'column' : 'row'
+                );
+            });
+
+            resizeObserver.observe(containerEl);
+        }
+
+        return () => {
+            if (resizeObserver !== null && containerEl) {
+                resizeObserver.unobserve(containerEl);
+            }
+        };
+    }, [setDisplayDirection, containerEl]);
```

Comment:
Why are we handling this? Was there a requirement that these display in this way?

Reply:
It was discussed in the running thread. I can let the options scrunch into themselves if you prefer.